GX-TimeFrequency/test/test_gx_minimax_grid.f90

```diff
@@ -44,7 +46,9 @@ program test_gx_minimax_grid
                          freq_mesh, freq_weights, &
                          cos_tau_to_freq_weights, cos_freq_to_tau_weights, &
                          sinft_tau_to_freq_weights, &
-                         max_errors, cosft_duality_error, ierr)
+                         max_errors, cosft_duality_error, ierr, &
+                         sinft_tw=sinft_freq_to_tau_weights, &             ! The sinft_tw weights are optional
```

Comment:
In general, I wouldn’t make any changes to this routine. The `ctest` is failing because it includes specific assertions that don’t account for the changes you implemented.

Instead of modifying this existing test, please create a new one, something similar to `gx_tabulate_minimax.F90`, for example `gx_tabulate_minimax_sine.F90`, where we can include the sine transformation and duality error checks.